registry/storage/shares.go

```diff
@@ -46,38 +47,116 @@ type Shares interface {
 }
 
 type sharesStorage struct {
-	logger *zap.Logger
-	db     basedb.Database
-	prefix []byte
-	shares map[string]*types.SSVShare
-	mu     sync.RWMutex
+	logger    *zap.Logger
+	db        basedb.Database
+	prefix    []byte
+	shares    map[string]*types.SSVShare
+	operators map[uint64][]byte
+	mu        sync.RWMutex
+}
+type storageOperator struct {
+	OperatorID spectypes.OperatorID
+	PubKey     []byte `ssz-size:"48"`
+}
+
+// Share represents a storage share.
+// The better name of the struct is storageShare,
+// but we keep the name Share to avoid conflicts with gob encoding.
+type Share struct {
```

Comment:
Was it created only because spec's Operator field names were changed which breaks SSZ encoding?

Reply:
The spec change caused a disruption in the database. Initially, we stored data using Share spectype, and altering the spec type resulted in a malfunction in the database. Now, we are managing storage using our custom mapped struct.

Reply:
ok, thanks